jupyter_core/utils/__init__.py

```diff
@@ -161,9 +160,11 @@ def wrapped(*args: Any, **kwargs: Any) -> Any:
             pass
 
         # Run the loop for this thread.
-        if name not in _loop_map:
-            _loop_map[name] = asyncio.new_event_loop()
-        loop = _loop_map[name]
+        try:
+            loop = asyncio.get_event_loop()
+        except RuntimeError:
```

Comment:
Do you actually want `loop = asyncio.get_running_loop()` here?

According to https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_event_loop this is deprecated in python 3.12 and will emit `https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_event_loop`. This causes some test failures (e.g. in jupyter_client).

Reply:
What I suggested above does fix the test failures in jupyter_client I was getting, at the expense of causing other test failures in jupyter_client.

The only thing that makes the jupyter_client testsuite pass (on python 3.12) is reverting the current PR. I'll downgrade to jupyter_core 5.3.1 for the time being.

Reply:
Thanks for the heads up @tornaria, I opened #367